### PR TITLE
[JENKINS-50061] Re-styled header

### PR DESCRIPTION
**Pull request by recena**
See [JENKINS-50061](https://issues.jenkins-ci.org/browse/JENKINS-50061).

- [ ] ~~Update JTH to use a more recent release of HtmlUnit [JENKINS-49491](https://issues.jenkins-ci.org/browse/JENKINS-49491)~~
- [x] Get a success execution of **Functional tests for Jenkins core**
- [ ] Get a success execution of **Acceptance test cases**

### Firefox on Mac

screenshot-ci jenkins io-2018-03-04-20-21-16

### Chrome on Mac

menubar_and_jenkins_and_re-styled_header_by_recena_ _pull_request__3285_ _jenkinsci_jenkins

menubar_and_dashboard__jenkins__and_layout-common_css_-_jenkins_-____development_projects_jenkins_

### Firefox on Mac

screenshot-localhost 8081-2018-03-04-16-37-13

screenshot-localhost 8081-2018-03-04-16-39-16

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests

### Desired reviewers

@jenkinsci/code-reviewers

### CODE REVIEW

File: core/src/main/resources/lib/layout/layout.jelly [hunk 1]
```diff
@@ -203,37 +201,26 @@ ${h.initPageVariables(context)}
...
-              <script>createSearchBox("${searchURL}");</script>
-            </div>
-          </form>
-        </div>
+          </li>
+          <li class="searchbox">
+            <j:set var="searchURL" value="${h.searchURL}"/>
+            <form action="${searchURL}" method="get" class="no-json" name="search">
+              <input name="q" placeholder="${%Search}" id="search-box" class="has-default-text" value="${request.getParameter('q')}" />
+              <!-- <div id="search-box-completion" /> -->
```
Comment (recena) [inline core/src/main/resources/lib/layout/layout.jelly]: This part is still in progress. I would like to make a proposal for removing this behavior.

**Comment by Wadeck** [comment]
#### Here are screenshots under Windows

<details><summary>Chrome</summary>

chrome_none
No security

chrome_anon
Anonymous user

chrome_admin
Logged in as admin
</details>

<details><summary>Firefox</summary>

firefox_none
No security

firefox_anon
Anonymous user

firefox_admin
Logged in as admin
</details>


<details><summary>Edge</summary>

edge_none
No security

edge_anon
Anonymous user

edge_admin
Logged in as admin
</details>


<details><summary>Internet Explorer 11</summary>

ie-11_none
No security

ie-11_anon
Anonymous user

ie-11_admin
Logged in as admin
</details>


#### Feedback:
- the new style is nice
- the distinction between sign-in icon and sign-up label is quite confusing, seems there is only one possible action
- the vertical align between icon + label is not consistent, but I know that's an NP-hard problem in browser compability science.
- ⚠️ under IE 11 the butler is broken (normally displayed in previous version)

**Comment by recena** [comment]
Awesome! I'm going to continue this weekend.

**Comment by daniel-beck** [comment]
The position of the admin monitor popup, as well as the 'arrow' between popup and button, seem off. During the animation, the are misaligned with each other as well.

**Comment by daniel-beck** [comment]
Also seems to remove the recently implemented admin monitor styling in the popup.

**Comment by recena** [comment]
@daniel-beck 

> Also seems to remove the recently implemented admin monitor styling in the popup.

Could you provide a screenshot? I'm not aware about it.

**Comment by daniel-beck** [comment]
PR build:

screen shot

2.107:

screen shot

**Comment by recena** [comment]
@daniel-beck In `2.105`

screenshot-localhost 8080-2018-02-19-23-49-47

Some alert box needs to be adapted in the main page.

**Comment by daniel-beck** [comment]
Looks like there's a systematic problem with the warning style. Or both info and danger. @recena Could you look into that?

# Manage Jenkins page

> screen shot

# Popup

> screen shot

**Comment by recena** [comment]
@daniel-beck Yes, I'm on it.

**Comment by recena** [comment]
@Wadeck IMHO the current implementation is a bit more robust regarding to the alignment.

File: test/src/test/java/hudson/model/ViewTest.java [hunk 2]
```diff
@@ -231,7 +231,7 @@
         ListView view = listView("v");
         view.description = "one";
         WebClient wc = j.createWebClient();
-        String xml = wc.goToXml("view/v/config.xml").getContent();
```
Comment (recena) [inline test/src/test/java/hudson/model/ViewTest.java]: Deprecated.

**Comment by recena** [comment]
This piece of code is very fragile:

```
<script><![CDATA[
var okButton = makeButton($('ok'),null);

function updateOk(form) {
  function state() {
    if($('name').value.length==0)   return true;
    var radio = form.elements['mode'];
    if (radio.length==2)  return false; // this means we only have dummy checkboxes
    for(i=0;i<radio.length;i++)
      if(radio[i].checked)
        return false;
    return true;
  }

  okButton.set('disabled',state(),false);
}
updateOk(okButton.getForm());
]]></script>
```

**Comment by Wadeck** [comment]
@recena after update the look & feel is even nicer! go on!

<details><summary>Screenshots Windows</summary>

Chrome without any security

chrome-none

Chrome with security enabled
chrome-login

Chrome logged in

chrome-logged

</details>

  
  
⚠️ Little proposal: increase the space between the user context menu arrow and the log out icon since at the moment there is no pixel of space between them, it's "easy" to misclick and so to disconnect instead of going to configure page.

<details><summary> 🐛 Bugs: signup</summary>

The signup button is lost in action. ✝️ 
chrome-signup-missing

Ho no, it was just hidden... it is a bit shy...
chrome-signup-found

</details>


The graphical bug I saw on IE is no more there, good job 👍 .
<details><summary>Screenshots Other</summary>

Firefox
ff-none

Edge
edge-none

Internet Explorer 11
ie-none


</details>

**Comment by Wadeck** [comment]
For the behavior close to the context menu:

context-logout

1. The arrow from the context menu stays during some seconds after we are out of its area, letting the user thinks it's still active.
2. As the logout icon clickable area is ~~too~~ (pixel perfectly) close, a miss-click is too easy. In terms of UX it's not good I think.

**Comment by recena** [comment]
@Wadeck What do you think?

screenshot-localhost 8081-2018-03-05-22-26-37

**Comment by Wadeck** [comment]
@recena your screenshot looks really good. I tested with other situation, other problems arise.

1. If you have warnings, they are not glued with the search bar, an inexperimented admin could imagine it's a feature linked with the search input.

recena-3-glued-warning


2. One logged in, as you put some space between the arrow (that is often invisible) and the log out icon, the username is not centered.

screenshot_2018-03-06_105009_001 - copie

As a proposal:
recena-3-proposal

To achieve this one:
1. add `margin-right: 6px` to `#visible-am-container` (for warnings)
2. add new rule `#header ul#toolbar li.username a`  with `padding-left:16px` inside (for centered username without touching the breadcrumb or other link in header)

**Comment by recena** [comment]
@Wadeck Great feedback. I'll address it.

**Comment by recena** [comment]
@Wadeck Addressed. What do you think?

**Review by Wadeck — APPROVED** [review]
@recena thank you for the modification, that's great :)
👍

**Comment by daniel-beck** [comment]
The smaller resolution of the title text image makes it look much worse when the web page is zoomed in, and, I assume, on Retina Macs. What motivates changing this resource file?

**Comment by recena** [comment]
@daniel-beck Could you provide a screenshot?

**Comment by recena** [comment]
@daniel-beck General speaking, I'm working, through several PRs, on replacing several old-resources by new ones.

**Comment by recena** [comment]
I think I would have to finish this proposal...

**Comment by oleg-nenashev** [comment]
It would be nice to get it landed, but it requires a merge conflict fix

**Comment by daniel-beck** [comment]
@recena Could you clarify whether you're still interested in getting this change merged?

**Comment by batmat** [comment]
We will likely close this on next pass in the next days if nothing is done in the meantime.

If so, please feel free anytime to get back to this and tell us so. We'll be happy to reopen as/if needed.

Thanks!

**Comment by recena** [comment]
Ok, I will file a new PR if needed.

**Comment by batmat** [comment]
Great! Thanks for the feedback. Closing this one then. Thanks again

**Pull request closed without merging**